presto-main/src/main/java/com/facebook/presto/operator/PartitionedLookupSourceFactory.java

```diff
@@ -381,15 +381,18 @@ private void freePartitions()
     private class SpillAwareLookupSourceProvider
             implements LookupSourceProvider
     {
+        private LookupSource lookupSource;
```

Comment:
See a comment in the code from somewhere in this file:
```
/*
 * lookupSourceSupplier exists so the now-spilled partition is still referenced by it. Need to re-create lookupSourceSupplier to let the memory go
 * and to prevent probe side accessing the partition.
 */
```

when a partition is spilled, the now-spilled partition would be held in this `lookupSource` until you clear it. But even if you consult spill epoch in `withLease` this won't necessarily be called soon. When memory is low, HBO memory may be revoked and LJO be stalled (blocked), so you can't assume it will dial here in next few seconds/minutes.